spring-cloud-function-context/src/test/java/org/springframework/cloud/function/cloudevent/CloudEventMessageUtilsAndBuilderTests.java

```diff
@@ -97,4 +97,45 @@ public void testAttributeRecognitionAndCanonicalization() {
 		assertThat(httpMessage.getHeaders().get("ce-specversion")).isNotNull();
 		assertThat(CloudEventMessageUtils.getSpecVersion(httpMessage)).isEqualTo("1.0");
 	}
+
+	@Test
+	void canonicalizeHeadersWithPossibleCopyReturnsCopyWithUpdatedHeadersWhenModified() {
+		// TODO add the following test cases
```

Comment:
I will do these in follow on PR if this proposal is accepted. I am out of juice for the day though.